localstack/services/s3/s3_utils.py

```diff
@@ -192,7 +192,7 @@ def authenticate_presign_url(method, path, headers, data=None):
                     elif key_lower in (blacklisted_header.lower() for blacklisted_header in BLACKLISTED_HEADERS):
                         pass
                     else:
-                        sign_headers[key] = query_params[key][0]
+                        query_string[key] = query_params[key][0]
```

Comment:
I had an issue in my lab when I tried this. Several tests that are using the V2 signature started failing.  Maybe it was just my lab but reverting that single change caused the tests to start working.